src/v/config/property.h

```diff
@@ -209,13 +254,14 @@ class property : public base_property {
     }
 
     T _value;
-    const T _default;
+    T _default;
+
+    // An alternative default that applies if the cluster's original logical
+    // version is <= the defined version
+    const std::optional<legacy_default<T>> _legacy_default;
```

Comment:
Doesn't have to be now, but I can imagine this eventually evolving into a list of defaults as we continue to tune certain defaults.

Reply:
Yeah, should be quick to add as/when we need it.